apps/protocol-frontend/src/components/MintModal.tsx

```diff
@@ -27,34 +29,46 @@ const MintModal = ({ contributions }: MintModalProps) => {
     }),
   );
   const [minting, setMinting] = useState(false);
-  const [mintProgress, setMintProgress] = useState(0);
-  const [mintTotal, setMintTotal] = useState(contributions?.length);
-
-  useEffect(() => {
-    if (minting && mintProgress === mintTotal) {
-      setMinting(false);
-    }
-  }, [mintProgress]);
 
   const mintHandler = async (contributions: MintContributionType[]) => {
-    setMintTotal(contributions.length);
     setMinting(true);
+    if (contributions.length > 1) {
+      const bulkStoreResult = await bulkStoreIpfs(
+        contributions.map(c => ({
+          content: {
+            name: c.original.name,
+            details: c.original.details,
+            proof: c.original.proof,
+          },
+        })),
+      );
+
+      // Mint successfully stored contributions in IPFS.
+      await bulkMintContributions(
+        bulkStoreResult
+          .filter(b => b.status === 'fulfilled')
+          .map(v => ({
+            ...contributions[v.index].original,
+            ipfsContentUri: v.value as string,
```

Comment:
```ts
          .filter(b => b.status === 'fulfilled')
          // ^ if it is fulfilled here, it must have a value but the type is not reflect: PromiseSettledResult<T>
          .map(v => ({
            ...contributions[v.index].original,
            ipfsContentUri: v.value as string,
          // ^ we are sure it's fulfilled, it must be a type of string.
          // otherwise, it will be `string | undefined`
          })),
```